.github/workflows/update-search.yml

```diff
@@ -21,13 +21,15 @@ jobs:
     - uses: actions/setup-node@v2
       with:
         node-version: 14
+    # configure AWS necessary for `nextstrain remote upload`
+    - uses: aws-actions/configure-aws-credentials@v1
+      with:
+        aws-region: ${{ secrets.AWS_DEFAULT_REGION }}
+        aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
     - uses: nextstrain/.github/actions/setup-nextstrain-cli@master
     - name: update sars-cov-2 search
       run: |
         npm ci
         ./scripts/collect-search-results.js --pathogen sars-cov-2
         nextstrain remote upload s3://nextstrain-data ./data/search_sars-cov-2.json
-      env:
-        AWS_DEFAULT_REGION: ${{ secrets.AWS_DEFAULT_REGION }}
-        AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
-        AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
```

Comment:
This change (and the corresponding one in the other workflow) increases the exposure of the AWS creds to more steps of the workflow. In particular, it exposes them to steps which install third-party software (via `setup-nextstrain-cli`), although they were already exposed to `npm ci`. I'd suggest splitting the steps and re-ordering so that setup/install stuff happens before the AWS creds are setup. More generally, I think that unlike `setup-node` and `setup-python` which we want to come early in a job, the AWS creds setup should come as late as possible.

Reply:
@tsibley good point, I've moved them.